mlrun/runtimes/pod.py

```diff
@@ -337,6 +341,27 @@ def with_node_selection(
         if affinity:
             self.spec.affinity = affinity
 
+    def with_priority_class(self, priority_class_name: typing.Optional[str]):
+        """
+        Enables to control the priority of the pod
+        If not passed - will default to the default_function_priority_class passed in config
+
+        :param priority_class_name:       The name of the priority class
+        """
+        valid_priority_class_names = self.list_valid_priority_class_names()
+        if priority_class_name not in valid_priority_class_names:
+            message = "Priority class name not in available priority class names"
+            logger.warning(
+                message,
+                priority_class_name=priority_class_name,
+                valid_priority_class_names=valid_priority_class_names,
+            )
+            raise mlrun.errors.MLRunInvalidArgumentError(message)
```

Comment:
So I see empty list means nothing is valid, commenting to remember to discuss on this

Reply:
Yes empty list means nothing is valid, we will use the default if present (added comment)